### PR TITLE
[v2.9] Update Rancher with new eks-operator version

### PR DESCRIPTION
**Pull request by mjura**
## Problem

eks-operator was updated to use new aws-sdk-go-v2 version. This adoption is impacting also Rancher integration with eks-operator. This change partially is fixing it.

 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,8 @@ replace (
 
 	github.com/knative/pkg => github.com/rancher/pkg v0.0.0-20190514055449-b30ab9de040e
 	github.com/matryer/moq => github.com/rancher/moq v0.0.0-20200712062324-13d1f37d2d77
+
+	github.com/rancher/eks-operator => github.com/mjura/eks-operator v1.3.1-mj-21032024-rc3
```
Comment (furkatgofurov7) [inline go.mod:14]: why are we using fork for replace?

Comment (mjura) [reply go.mod]: fixed

**Review by salasberryfin — APPROVED** [review]
LGTM. https://github.com/rancher/eks-operator/pull/426 should be merged before this, shouldn't it?

File: pkg/controllers/management/eks/eks_cluster_handler.go [hunk 2]
```diff
@@ -587,7 +623,7 @@ func (e *eksOperatorController) getRestConfig(cluster *mgmtv3.Cluster, dialer ty
 
 func notFound(err error) bool {
 	if awsErr, ok := err.(awserr.Error); ok {
-		return awsErr.Code() == eks.ErrCodeResourceNotFoundException
+		return awsErr.Code() == "ResourceNotFoundException"
```
Comment (yiannistri) [inline pkg/controllers/management/eks/eks_cluster_handler.go:626]: What is the downside of using a [const](https://github.com/aws/aws-sdk-go/blob/main/service/eks/errors.go#L77)  here?

**Review by yiannistri — DISMISSED** [review]

**Review by salasberryfin — APPROVED** [review]

File: pkg/controllers/management/clusterupstreamrefresher/eks_upstream_spec.go [hunk 3]
```diff
@@ -66,21 +67,25 @@ func BuildEKSUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgmtv3.C
 		if len(ng.Nodegroup.Health.Issues) != 0 {
 			var issueMessages []string
 			for _, issue := range ng.Nodegroup.Health.Issues {
-				issueMessages = append(issueMessages, aws.StringValue(issue.Message))
-				if !ekscontroller.NodeGroupIssueIsUpdatable(aws.StringValue(issue.Code)) {
+				issueMessages = append(issueMessages, aws.ToString(issue.Message))
+				if !ekscontroller.NodeGroupIssueIsUpdatable(string(issue.Code)) {
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply pkg/controllers/management/clusterupstreamrefresher/eks_upstream_spec.go:71]: This was resolved without accepting the change?

**Review by furkatgofurov7 — APPROVED** [review]
Only one nit, not a blocker most probably

**Pull request merged**